drivers/flash/flash_mcux_flexspi_nor.c

```diff
@@ -241,7 +241,7 @@ static int flash_flexspi_nor_erase_sector(const struct device *dev,
 		.dataSize = 0,
 	};
 
-	LOG_DBG("Erasing sector at 0x%08x", offset);
```

Comment:
Maybe rather "z' length modifier should be used with "x" and the `offset` should be cast to `ssize_t`.
We should also consider adding  offset specific formatting to the lib/libc/minimal/include/inttypes.h.

Reply:
I thought that casting into unsigned int would be enough here. If we really want to change it into `ssize_t` we also have to fix it in `eeprom_lpc11u6x` where the same casting is being done. The last option with adding offset type into `inttype.h` IMO is the best one.

Reply:
> 
> I thought that casting into unsigned int would be enough here.

`ssize_t` would be probably closest to `off_t` in size and it allows to use `z` modifier with x. Casting to int may be truncation for systems that have 64bit off_t and 32 bit int.

> The last option with adding offset type into inttype.h IMO is the best one.

You may RFC such change.

Reply:
>The ssize_t would be probably closest to the off_t in size and it allows to use z modifier with x. Casting to int may be truncation for systems that have 64bit off_t and 32 bit int.

Okay, I'll change that into ssize_t then.

>You may RFC such change.

I'm gonna put this off for a while since I'm not sure about which exactly specifier should be used for off_t type. If you have some idea I would be glad ;)

Reply:
@de-nordic replaced with ssize_t.

Reply:
Please set DNM label, this change causes shell deadlock.